AGENTS.md

```diff
@@ -0,0 +1,143 @@
+This file provides guidance to AI agents when working with code in this repository.
+
+This is the OpenShift API repository - the canonical location of OpenShift API type definitions and serialization code. It contains:
+
+- API type definitions for OpenShift-specific resources (Custom Resource Definitions)
+- FeatureGate management system for controlling API availability across cluster profiles
+- Generated CRD manifests and validation schemas
+- Integration test suite for API validation
+
+## Key Architecture Components
+
+### FeatureGate System
+The FeatureGate system (`features/features.go`) controls API availability across different cluster profiles (Hypershift, SelfManaged) and feature sets (Default, TechPreview, DevPreview). Each API feature is gated behind a FeatureGate that can be enabled/disabled per cluster profile and feature set.
+
+### API Structure
+APIs are organized by group and version (e.g., `route/v1`, `config/v1`). Each API group contains:
+- `types.go` - Go type definitions
+- `zz_generated.*` files - Generated code (deepcopy, CRDs, etc.)
+- `tests/` directories - Integration test definitions
+- CRD manifest files
+
+## Common Development Commands
+
+### Building
+```bash
+make build              # Build render and write-available-featuresets binaries
+make clean              # Clean build artifacts
+```
+
+### Code Generation
+```bash
+make update             # Alias for update-codegen-crds
+```
+
+### Testing
+```bash
+make test-unit          # Run unit tests
+make integration        # Run integration tests (in tests/ directory)
```

Comment:
Should we teach it how to run the integration tests with more focused arguments?

That way running the integration tests don't take longer than necessary when reviewing a change that only impacts a subset of the APIs/tests?